File: python/calculate_mode.py
def calculate_mode(input_arr):
    answer=[]
    obj={}
    for value in input_arr:
        if value in obj.keys():
            obj[value]+=1
        else:
            obj[value]=1
    most_rep = max(obj, key=obj.get)
    for key in obj:
        if obj[key] >= obj[most_rep]:
            answer.append(key)
    return answer

File: js/calculateMode.js
const calculateMode = (inputArr) => {
    let answer=[]
    let obj={}
    let mostRep=inputArr[0];
    inputArr.forEach(value => {
        if (obj[value]){
        obj[value] +=1
    }else{
        obj[value]=1
    }})
    for (let key in obj)
        if (obj[key] > obj[mostRep]){
            mostRep = key;
        }
    for (let key in obj){
        if (obj[key] >= obj[mostRep]){
            answer.push(key)
    }}
    return answer
}

